Modules/Forum/classes/class.ilForumPostDraft.php

```diff
@@ -384,7 +387,52 @@ protected static function readDrafts($user_id)
             self::$instances[$user_id]['draft_ids'][$tmp_obj->getDraftId()] = $tmp_obj;
         }
     }
+
+    /**
+     * @param int $user_id
+     * @param int $thread_id
+     * @param int $sorting
+     * @return ilForumPostDraft[]
+     */
+    public static function readSortedDrafts(int $user_id, int $thread_id, int $sorting = ilForumProperties::VIEW_DATE_ASC) : array
+    {
+        global $DIC;
+        $ilDB = $DIC->database();
+    
+        $drafts = [];
+    
+        $order_statement = ' ';
+        $order_direction = ' ';
     
+        if ($sorting == ilForumProperties::VIEW_DATE_ASC || $sorting == ilForumProperties::VIEW_DATE_DESC) {
```

Comment:
@nmatuschek Can't we use a default sorting and use only one `if` statement to adust the sorting in this case?

Reply:
Yes,  we could use if($sorting != ilForumProperties::VIEW_TREE)

Reply:
Okay, please try to improve this. Thx in advance.